express-notes-api/server.js

```diff
@@ -0,0 +1,119 @@
+import express from 'express';
+import { writeFile, readFile } from 'node:fs/promises';
+const app = express();
+const port = 8080;
+
+async function getFiles() {
+  try {
+    const filePath = new URL('data.json', import.meta.url);
+    const content = await readFile(filePath, 'utf8');
+    return content;
+  } catch (error) {
+    console.error('Error');
+  }
+}
+
+async function processFiles(url, content) {
+  try {
+    await writeFile('data.json', content);
+    console.log(content);
+  } catch (error) {
+    console.error('Error');
+  }
+}
+
+var data = JSON.parse(await getFiles('data.json'));
+
+app.use(express.json());
+
+// Clients can GET a list of notes
+app.get('/api/notes', async (req, res) => {
+  try {
+    res.json(Object.values(data.notes));
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+// Client can GET a note with ID
+app.get('/api/notes/:id', (req, res) => {
+  const findRecord = req.params.id;
+  const matchingID = data.notes[findRecord];
+
+  if (findRecord <= 0 || !Number.isInteger(Number(findRecord))) {
+    res.status(400).json({ error: 'id must be a positive interger' });
+  } else if (!data.notes[findRecord]) {
+    res.status(200).json({ error: `cannot find note with id ${findRecord}` });
```

Comment:
If a note is missing, the server should respond with a `404 Not Found`:

---

![image](https://github.com/rachel-lyy/m2-0323-code-solutions/assets/16143771/60832a74-64ee-42a6-9b8d-f9cae11656ac)